### ISSUE TITLE
[Feat Request] subpackage python folder conflict with subpackage python file

### ISSUE THREAD

**Issue by moylop260** [PyCQA/pylint#1038]
### Steps to reproduce
1. Create the following directory structure:

``` bash
 ./my_wrong_package
    ./my_wrong_subpackage.py
    ./my_wrong_subpackage/
        __init__.py
    __init__.py
```

Or run the following script:

``` bash
mkdir -p ./my_wrong_package/my_wrong_subpackage/
touch ./my_wrong_package/{.,my_wrong_subpackage}/__init__.py
touch ./my_wrong_package/my_wrong_subpackage.py
```
### Proposal behaviour

My proposal is emit a message when a folder python have the same name that a
python file in the same structure of directories.

Is a good idea for pylint?

**Comment by moylop260** [comment]
@PCManticore 
What are you think about?

**Comment by PCManticore** [comment]
Not sure this is actually worth it.
